public/components/trace_analytics/requests/request_handler.ts

```diff
@@ -27,6 +27,24 @@ export function handleDslRequest(http: CoreStart['http'], DSL: any, bodyQuery: a
   if (!bodyQuery.index) {
     body = {...bodyQuery, index: ((mode === 'jaeger' ? JAEGER_INDEX_NAME : DATA_PREPPER_INDEX_NAME)) }
   }
+  if (mode === 'jaeger' && timeout) {
+    const controller = new AbortController();
+    const id = setTimeout(() => controller.abort(), 20000);
```

Comment:
you can show request taking long time toast here (or trigger an callback) and remove abortcontroller. i don't think we should abort the request in 20s, it's limiting the user